add zero-input-panel-posframe package.

sylecn opened the pull request:
### Brief summary of what the package does
zero-input is a Chinese input method framework written as Emacs minor mode. This package provides an optional panel based on posframe package. A panel is used to show candidates for zero-input methods.

### Direct link to the package repository

https://gitlab.emacsos.com/sylecn/zero-input-panel-posframe.git

### Your association with the package

I am the author and maintainer of this new package and the original zero-input package, which this package depends on.

### Relevant communications with the upstream package maintainer

**None needed**

### Checklist

<!-- Please confirm by replacing `[]` with `[x]`: -->

- [x] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses)
- [x] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [x] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [x] My elisp byte-compiles cleanly
- [x] I've used `M-x checkdoc` to check the package's documentation strings
- [x] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)

<!-- After submitting, please fix any problems the CI reports. -->

riscy commented:
Thanks for this.  Quick first pass --

`zero-input-panel-posframe.el` with byte-compile using Emacs 29.3:
```
zero-input-panel-posframe.el:42:2: Error: Symbol's function definition is void: ert-set-test
```
This looks like an issue with the dependency on `zero-input` - and what appears to be happening is that some chain of events is leading to test code being brought in (typically MELPA excludes test code from packages). I think since zero-input is also your package, we should wait on merging *this* package until that issue is resolved.

Package and license:
- Avoid specifying `:branch` except in unusual cases.  If `master` is your default branch, you don't need to specify it in your recipe

sylecn commented:
Hi riscy,

I have removed the ert test from zero-input.el (from git master and lastest tag v2.10.3). That should fix the byte-compile issue on ert-set-test.

dropped ":branch master" in new pkg recipe as you suggested.


There are a few warnings I believe is false positive (in case you notice them):
1. checkdoc on zero-input.el: symbol zero-input-candidates-per-page not quoted. This is expected, because it's an s-exp user can copy and paste to init.el file. If I escape it with quotes, the rendered document will be not copy-paste-able.

This is the relevant code:
```elisp
(defvar-local zero-input-initial-fetch-size 21
  "How many candidates to fetch for the first call to GetCandidates.

It\\='s best set to \\=(1+ (* zero-input-candidates-per-page N)) where
N is number of pages you want to fetch in initial fetch.")
```
2. when install zero-input package, byte-compile warnings on focus-in-hook, focus-out-hook being obsolete variable. This is handled in elisp code, in order to support older version of emacs.

3. when install zero-input package, "`zero-input-pinyin-fuzzy-flag' docstring wider than 80 characters", that is because there is a file path and a URL in the docstring, I am not sure how I can make the warning go without losing the information. In source code, I already used \ to escape the long line, so source code is not wider than 80 chars.

Here is the relevant code:
```elisp
(defcustom zero-input-pinyin-fuzzy-flag 0
  "Non-nil means enable fuzzy pinyin when calling zero pinyin service.

Fuzzy pinyin means some shengmu and some yunmu could be used
interchangeably, such as zh <-> z, l <-> n.

For supported values, please see zero-input-pinyin-service dbus
interface xml comment for GetCandidatesV2 method fuzzy_flag param.

You can find the xml file locally at
/usr/share/dbus-1/interfaces/\
com.emacsos.zero.ZeroPinyinService1.ZeroPinyinServiceInterface.xml
or online at
https://gitlab.emacsos.com/sylecn/zero-pinyin-service/\
blob/master/com.emacsos.zero.ZeroPinyinService1.ZeroPinyinServiceInterface.xml"
  :group 'zero-input-pinyin
  :type 'integer)
```

riscy commented:
Yes, these are fine - checkdocs are these days "within reason" and there are no critical ones I can see.  Merging - and thanks for the fixes!
